### ISSUE TITLE
Code samples get formatted incorrectly

### ISSUE THREAD

**Issue by G4Vi** [dylibso/xtp-python-bindgen#18]
In mcp.run the schema has the code sample as such and it looks correctly formatted:
```yaml
- lang: python
        label: Description of a simple greeting tool that greets someone by name
        source: |
          name = input.params.name
          if name is None:
              raise Exception("Argument `name` must be provided")
          return CallToolResult(
              content=[
                  Content(
                      text=f'Hello, {name}',
                      mimeType="text/plain",
                      type=ContentType.Text,
                      annotations=None,
                      data=None,
                  )
              ],
              isError=False,
          )
```

After `xtp plugin init --feature stub-with-code-samples`,
the python has incorrect indentation and does not build.
```python
def call(input: CallToolRequest) -> CallToolResult:
    name = input.params.name


if name is None:
    raise Exception("Argument `name` must be provided")
return CallToolResult(
    content=[
        Content(
            text=f"Hello, {name}",
            mimeType="text/plain",
            type=ContentType.Text,
            annotations=None,
            data=None,
        )
    ],
    isError=False,
)
```

I did not have to reformat the `describe` export.

**Comment by nilslice** [comment]
Good catch -- its hard to format stuff in YAML and expect it to persist in all the right ways when we inject it into the templates.. I wonder though, why the format script doesn't handle this. Looks like it is indeed filled in: 
https://github.com/dylibso/xtp-python-bindgen/blob/14a4455d7b15102e1ce3e0640422e7502f124680/template/xtp.toml.ejs#L14

@G4Vi can you tell if that script failed to run or reported an error or something?

**Comment by G4Vi** [comment]
> @G4Vi can you tell if that script failed to run or reported an error or something?

No error initializing the plugin from what I can see:
```
gavin@Gavins-MacBook-Pro servlets % xtp plugin init --feature stub-with-code-
samples
Enabling flag: stub-with-code-samples
Generating plugin for Tool in abdef...
👉 Running prepare script: bash prepare.sh && uv sync
Using CPython 3.13.0
Creating virtual environment at: .venv
Resolved 5 packages in 96ms
Installed 4 packages in 4ms
 + nodeenv==1.9.1
 + pyright==1.1.391
 + ruff==0.8.3
 + typing-extensions==4.12.2
👉 Running format script: uv run ruff format plugin/*.py
4 files reformatted
✅ Generated plugin for Tool in abdef
```

For good measure I ran
```
gavin@Gavins-MacBook-Pro abdef % uv run ruff format plugin/*.py
4 files left unchanged
```
and that also did not error and returned a successful exit code.

**Comment by nilslice** [comment]
> For good measure I ran
> 
> ```
> gavin@Gavins-MacBook-Pro abdef % uv run ruff format plugin/*.py
> 4 files left unchanged
> ```
> 
> manually after init and that also did not error and returned a successful exit code.

Hm! And doing so didn't format that file I'm guessing?

**Comment by G4Vi** [comment]
> Hm! And doing so didn't format that file I'm guessing?

Yes, did not make any changes to the file.

**Comment by bhelx** [comment]
i suspect because this is technically valid python code. tabs have semantic meaning :(

**Comment by nilslice** [comment]
Ok I have repro'd this too. Strange. I'm not sure that the formatter detects anything wrong with the file, since it seems to only work on the file when I make manual changes to that file first to format it more "correctly"?

**Comment by nilslice** [comment]
I also changed the indentation in the schema, and now it's not working, but in a more expected way... so that's good?

**Comment by nilslice** [comment]
I think the issue is that the YAML schema doesn't preserve tabs, only uses spaces. But, when it reaches the template, the initial indentation is a tab: 
https://github.com/dylibso/xtp-python-bindgen/blob/14a4455d7b15102e1ce3e0640422e7502f124680/template/plugin/plugin.py.ejs#L18-L20

There seems to be an additional issue when the sample is inserted that throws off the formatter too.

**Comment by bhelx** [comment]
>  I'm not sure that the formatter detects anything wrong with the file

i think the bad code is syntactically *correct*.  but, because inserting whitespace into the python code changes the semantic meaning. it doesn't know what to format it to. it's not like a normal programming language where whitespace doesn't mean anything and you can format it however you want and the ast doesn't change. it might be that this strategy of expecting the "formatter to fix it" just wont work for python. 

@zshipko must have run into this when generating the code. maybe he has some insight.

**Comment by zshipko** [comment]
ah yeah, this is a rough edge with the python bindgen code samples, I think the best way to fix this for now is to add the indentation to the sample code - I will open up a PR for that

**Comment by zshipko** [comment]
actually changing it in the schema is too hacky - maybe @G4Vi can fix this at the template level instead while cleaning up the python bindgen

**Comment by G4Vi** [comment]
> maybe @G4Vi can fix this at the template level instead while cleaning up the python bindgen

Thanks for the pointer, I'll take a look template.

**Comment by G4Vi** [comment]
Fixed in #20
